traffic_ops/app/lib/API/DeliveryService/KeysUrlSig.pm

```diff
@@ -26,6 +26,32 @@ use constant URL_SIG_KEYS_BUCKET => "url_sig_keys";
 use Exporter qw(import);
 our @EXPORT_OK = qw(URL_SIG_KEYS_BUCKET);
 
+
+sub view_by_id {
+	my $self                = shift;
+	my $id              = $self->param('id');
+	my $rs = $self->db->resultset("Deliveryservice")->find( { id => $id } ); 
+	my $xml_id;
+	if ( defined($rs) ) {
+		$xml_id = $rs->xml_id;
+	}
+	else {
+		return $self->alert("Delivery Service '$id' does not exist.");
+	}
+
```

Comment:
can you add a tenant check in here similar to this:

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/API/Deliveryservice.pm#L175
 
going forward, anything ds related needs to check that the user requesting the ds info has the proper tenant to do so.